crates/typst/src/math/underover.rs

```diff
@@ -267,30 +397,28 @@ fn layout_underoverspreader(
     let glyph = GlyphFragment::new(ctx, styles, c, span);
     let stretched = glyph.stretch_horizontal(ctx, body.width(), Abs::zero());
 
-    let mut rows = vec![MathRun::new(vec![body]), stretched.into()];
-
-    let (sup_style, sub_style);
-    let row_styles = if reverse {
-        sup_style = style_for_subscript(styles);
-        styles.chain(&sup_style)
-    } else {
-        sub_style = style_for_subscript(styles);
-        styles.chain(&sub_style)
+    let mut rows = vec![MathRun::new(vec![body])];
+    let baseline = match position {
+        Position::Under => {
+            rows.push(stretched.into());
+            if let Some(annotation) = annotation {
+                let under_style = style_for_subscript(styles);
+                let annotation_styles = styles.chain(&under_style);
+                rows.push(ctx.layout_into_run(annotation, annotation_styles)?);
+            }
+            0
+        }
+        Position::Over => {
+            rows.insert(0, stretched.into());
+            if let Some(annotation) = annotation {
+                let over_style = style_for_superscript(styles);
+                let annotation_styles = styles.chain(&over_style);
+                rows.insert(0, ctx.layout_into_run(annotation, annotation_styles)?);
+            }
```

Comment:
Somewhat contrary to my comment in https://github.com/typst/typst/pull/3869, I think we should actually just use `push` and move the `MathRun::new(vec![body])` into the match as well, instead of doing `insert(0, ..)` twice.